Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.Cursor.cs

```diff
@@ -0,0 +1,225 @@
+        
+        // <summary>
+        // Gets or sets a cursor object that represents the mouse cursor.
+        // </summary>
+        // 
+        // <returns>
+        // A <see cref="T:System.Windows.Forms.Cursor"/> that represents the mouse cursor. The default is null if the mouse cursor is not visible.
+        // </returns>
+        // 
+        // public static Cursor Current {get; set;}
+        
+        // <summary>
+        // Gets the handle of the cursor.
+        // </summary>
+        // 
+        // <returns>
+        // An <see cref="T:System.IntPtr"/> that represents the cursor's handle.
+        // </returns>
+        // <exception cref="T:System.Exception">The handle value is <see cref="F:System.IntPtr.Zero"/>. </exception>
+        // public IntPtr Handle {get;}
+        
+        // <summary>
+        // Gets the cursor hot spot.
+        // </summary>
+        // 
+        // <returns>
+        // A <see cref="T:System.Drawing.Point"/> representing the cursor hot spot.
+        // </returns>
+        // public Point HotSpot { get; }
+        
+        // <summary>
+        // Gets or sets the cursor's position.
+        // </summary>
+        // 
+        // <returns>
+        // A <see cref="T:System.Drawing.Point"/> that represents the cursor's position in screen coordinates.
+        // </returns>
+        // public static Point Position {get; set;}
+        
+        // <summary>
+        // Gets the size of the cursor object.
+        // </summary>
+        // 
+        // <returns>
+        // A <see cref="T:System.Drawing.Size"/> that represents the width and height of the <see cref="T:System.Windows.Forms.Cursor"/>.
+        // </returns>
+        // public Size Size {get;}
+        
+        // <summary>
+        // Gets or sets the object that contains data about the <see cref="T:System.Windows.Forms.Cursor"/>.
+        // </summary>
+        // 
+        // <returns>
+        // An <see cref="T:System.Object"/> that contains data about the <see cref="T:System.Windows.Forms.Cursor"/>. The default is null.
+        // </returns>
+        // public object Tag {get; set;}
+        
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T:System.Windows.Forms.Cursor"/> class from the specified Windows handle.
+        /// </summary>
+        /// <param name="handle">An <see cref="T:System.IntPtr"/> that represents the Windows handle of the cursor to create. </param><exception cref="T:System.ArgumentException"><paramref name="handle"/> is <see cref="F:System.IntPtr.Zero"/>. </exception>
+        public Cursor(IntPtr handle)
+        {
+            Contract.Requires(handle != IntPtr.Zero);
+        }
+
+        // <summary>
+        // Initializes a new instance of the <see cref="T:System.Windows.Forms.Cursor"/> class from the specified file.
+        // </summary>
+        // <param name="fileName">The cursor file to load. </param>
+        // public Cursor(string fileName)
```

Comment:
We can add requires for this constructor:

`Contract.Requires(!string.IsNullOrEmpty(fileName))`

Constructor itself does not have this check, but it calls FileStream that do have this check in the `Init` method.